shared/actions.ts

```diff
@@ -0,0 +1,57 @@
+import { Upstream } from './state';
```

Comment:
would it make sense to move the new `shared` and `store` folders into `preview-src` since they're being used by the webview?

Reply:
Nevermind, I see that we import from `store` on the extension side